// animal-island-ui/dist/es/components/Table/Table.js
import { jsxs as c, jsx as l } from "react/jsx-runtime";
import t from "./table.module.less.js";
const j = ({
  columns: n = [],
  dataSource: d = [],
  rowKey: h = "key",
  striped: p = !0,
  showHeader: f = !0,
  rowClassName: r,
  onRow: m,
  loading: o = !1,
  emptyText: v = "暂无数据",
  scroll: y,
  className: g,
  style: N
}) => {
  const b = (e, a) => typeof h == "function" ? h(e) : e[h] || String(a), u = (e, a) => {
    const s = [t.row];
    return p && a % 2 === 1 && s.push(t.striped), r && (typeof r == "function" ? s.push(r(e, a)) : s.push(r)), s.join(" ");
  }, w = (e, a, s) => {
    const i = e.dataIndex ? a[e.dataIndex] : void 0;
    return e.render ? e.render(i, a, s) : i;
  }, x = [t.table, o && t.loading, g].filter(Boolean).join(" "), C = [t.wrapper, y && t.scrollable].filter(Boolean).join(" ");
  return /* @__PURE__ */ c("div", { className: C, style: N, children: [
    /* @__PURE__ */ c("table", { className: x, children: [
      f && /* @__PURE__ */ l("thead", { className: t.thead, children: /* @__PURE__ */ l("tr", { className: t.headerRow, children: n.map((e, a) => /* @__PURE__ */ l(
        "th",
        {
          className: t.headerCell,
          style: {
            width: e.width,
            textAlign: e.align || "left",
            ...e.style
          },
          children: e.title
        },
        a
      )) }) }),
      /* @__PURE__ */ l("tbody", { className: t.tbody, children: d.length === 0 ? /* @__PURE__ */ l("tr", { children: /* @__PURE__ */ l("td", { colSpan: n.length, className: t.emptyCell, children: /* @__PURE__ */ c("div", { className: t.emptyContent, children: [
        /* @__PURE__ */ l("svg", { className: t.emptyIcon, viewBox: "0 0 24 24", width: "48", height: "48", children: /* @__PURE__ */ l(
          "path",
          {
            fill: "currentColor",
            d: "M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM7 10h2v7H7zm4-3h2v10h-2zm4 6h2v4h-2z"
          }
        ) }),
        /* @__PURE__ */ l("span", { children: v })
      ] }) }) }) : d.map((e, a) => /* @__PURE__ */ l(
        "tr",
        {
          className: u(e, a),
          ...m?.(e, a),
          children: n.map((s, i) => /* @__PURE__ */ l(
            "td",
            {
              className: t.cell,
              style: {
                textAlign: s.align || "left",
                ...s.style
              },
              children: w(s, e, a)
            },
            i
          ))
        },
        b(e, a)
      )) })
    ] }),
    o && /* @__PURE__ */ l("div", { className: t.loadingOverlay, children: /* @__PURE__ */ l("div", { className: t.loadingSpinner, children: /* @__PURE__ */ l("svg", { viewBox: "0 0 50 50", width: "40", height: "40", children: /* @__PURE__ */ l(
      "circle",
      {
        cx: "25",
        cy: "25",
        r: "20",
        fill: "none",
        stroke: "currentColor",
        strokeWidth: "4",
        strokeLinecap: "round",
        strokeDasharray: "31.4 31.4"
      }
    ) }) }) })
  ] });
};
j.displayName = "Table";
export {
  j as Table
};
