// animal-island-ui/dist/es/components/Table/table.module.less.js
import "./table.module.css";
const a = "animal-wrapper-LJBly", l = "animal-scrollable-5Wnhh", e = "animal-table-Os4fM", n = "animal-thead-2ge5M", t = "animal-headerRow-sAsWX", o = "animal-headerCell-LhL6h", i = "animal-tbody-3RGsp", s = "animal-row-iDOMw", r = "animal-cell-4PAU2", c = "animal-striped-8Ih-N", m = "animal-emptyCell--stiv", d = "animal-emptyContent-JrC1j", p = "animal-emptyIcon-SNcRm", y = "animal-loading-mPt0l", h = "animal-loadingOverlay-PMW6Z", g = "animal-loadingSpinner-KZEsv", w = {
  wrapper: a,
  scrollable: l,
  table: e,
  thead: n,
  headerRow: t,
  headerCell: o,
  tbody: i,
  row: s,
  cell: r,
  striped: c,
  emptyCell: m,
  emptyContent: d,
  emptyIcon: p,
  loading: y,
  loadingOverlay: h,
  loadingSpinner: g
};
export {
  r as cell,
  w as default,
  m as emptyCell,
  d as emptyContent,
  p as emptyIcon,
  o as headerCell,
  t as headerRow,
  y as loading,
  h as loadingOverlay,
  g as loadingSpinner,
  s as row,
  l as scrollable,
  c as striped,
  e as table,
  i as tbody,
  n as thead,
  a as wrapper
};
