// animal-island-ui/dist/es/components/Form/FormItem.js
import { jsx as v, jsxs as m, Fragment as ge } from "react/jsx-runtime";
import M, { useContext as ye, useMemo as be, useState as ve, useCallback as Ce, useEffect as j } from "react";
import C from "classnames";
import r from "./Form.module.less.js";
import { FormContext as $e } from "./context.js";
import { stringifyNamePath as xe } from "./types.js";
function Fe(s) {
  if (s == null || typeof s != "object") return s;
  const e = s.target;
  if (e && typeof e == "object") {
    if ((e.type === "checkbox" || e.type === "radio") && "checked" in e)
      return e.checked;
    if ("value" in e && e.value !== void 0) return e.value;
  }
  return "value" in s ? s.value : s;
}
const o = "island-form-item", Ae = (s) => {
  const {
    name: e,
    label: p,
    rules: d = [],
    required: L = !1,
    valuePropName: P = "value",
    trigger: $ = "onChange",
    getValueFromEvent: z = Fe,
    normalize: x,
    hidden: T = !1,
    hasFeedback: _ = !1,
    validateStatus: I,
    help: G,
    noStyle: W = !1,
    labelCol: D,
    wrapperCol: H,
    colon: K,
    requiredMark: B,
    layout: J,
    className: O,
    children: u
  } = s, F = ye($e);
  if (!F)
    throw new Error("Form.Item must be used inside <Form> or <Form.Provider>");
  const {
    form: i,
    prefixCls: k,
    layout: Q,
    labelAlign: U,
    labelCol: X,
    wrapperCol: Y,
    size: N,
    disabled: Z,
    colon: ee,
    requiredMark: re
  } = F, t = be(() => e !== void 0 ? xe(e) : null, [e]), [, oe] = ve(0), V = Ce(() => oe((l) => l + 1), []);
  if (j(() => {
    if (!t) return;
    const n = i.__store;
    if (n)
      return n.registerField(e, d, void 0, V), () => n.unregisterField(t);
  }, [i, t, e, V]), j(() => {
    if (!t) return;
    i.__store?.updateRules?.(t, d);
  }, [d, t, i]), T) return null;
  const te = t ? i.getFieldValue(e) : void 0, h = t ? i.getFieldError(e) : void 0, le = t ? i.isFieldValidating(e) : !1, se = t ? i.isFieldTouched(e) : !1, c = I ?? (le ? "validating" : h?.[0] ? "error" : ""), q = (se && h?.[0] ? h[0] : void 0) ?? G, ie = B ?? re, w = (L || d.some((l) => typeof l == "object" ? l.required : !1)) && ie !== !1, g = M.isValidElement(u), a = g ? { ...u.props } : {};
  if (t && g) {
    a[P] = te;
    const l = a[$];
    a[$] = (n) => {
      typeof l == "function" && l(n);
      const f = z(n), b = i.getFieldValue(e), he = x ? x(f, b, i.getFieldsValue(!0)) : f;
      i.setFieldValue(e, he);
    };
  }
  Z && a.disabled === void 0 && (a.disabled = !0), a.size === void 0 && (a.size = N), a.status === void 0 && c === "error" && (a.status = "error");
  const E = t && g ? M.cloneElement(u, a) : u, R = J ?? Q, ae = R === "inline" ? "vertical" : R, y = D ?? X, ne = H ?? Y, ce = K ?? ee, S = (l, n = 1) => {
    if (!l) return {};
    const f = l.span ?? 24, b = l.offset ?? 0;
    return {
      gridColumn: `${n + b} / span ${f}`
    };
  }, de = (y?.span ?? 0) + (y?.offset ?? 0), ue = S(y, 1), fe = S(ne, de + 1), me = C(
    k ? r[`${k}-item`] : r[o],
    r[`${o}-${ae}`],
    r[`${o}-${N}`],
    {
      [r[`${o}-has-error`]]: c === "error",
      [r[`${o}-has-warning`]]: c === "warning",
      [r[`${o}-has-success`]]: c === "success",
      [r[`${o}-is-validating`]]: c === "validating",
      [r[`${o}-required`]]: w
    },
    O
  ), pe = p !== void 0 ? /* @__PURE__ */ v(
    "label",
    {
      htmlFor: t ?? void 0,
      className: C(r[`${o}-label`], {
        [r[`${o}-label-required`]]: w,
        [r[`${o}-label-colon`]]: ce && p !== ""
      }),
      style: { ...ue, textAlign: U },
      children: p
    }
  ) : null, A = q !== void 0 ? /* @__PURE__ */ m(
    "div",
    {
      className: C(r[`${o}-explain`], {
        [r[`${o}-explain-error`]]: c === "error"
      }),
      children: [
        _ && c === "error" ? /* @__PURE__ */ v("span", { className: r[`${o}-feedback-icon`], children: "✕" }) : null,
        q
      ]
    }
  ) : null;
  return W ? /* @__PURE__ */ m(ge, { children: [
    E,
    A
  ] }) : /* @__PURE__ */ m("div", { className: me, "data-field-name": t ?? void 0, children: [
    pe,
    /* @__PURE__ */ m("div", { className: r[`${o}-control`], style: fe, children: [
      /* @__PURE__ */ v("div", { className: r[`${o}-control-input`], children: E }),
      A
    ] })
  ] });
};
export {
  Ae as FormItem
};
