// animal-island-ui/dist/es/components/Form/Form.js
import { jsx as h } from "react/jsx-runtime";
import E, { useRef as x, useEffect as b, useMemo as I } from "react";
import J from "classnames";
import f from "./Form.module.less.js";
import { FormContext as N } from "./context.js";
import { useForm as A } from "./useForm.js";
import { FormItem as K } from "./FormItem.js";
const t = "island-form";
function B(s, F) {
  const {
    form: i,
    initialValues: l,
    layout: n = "horizontal",
    labelAlign: y = n === "horizontal" ? "right" : "left",
    labelCol: C,
    wrapperCol: P,
    size: p = "middle",
    disabled: v = !1,
    colon: R = !0,
    requiredMark: g = !1,
    onFinish: a,
    onFinishFailed: m,
    onValuesChange: c,
    onReset: $,
    className: z,
    children: _,
    ...D
  } = s, M = i !== void 0, [S] = A(), r = M ? i : S, d = x({ onFinish: a, onFinishFailed: m, onValuesChange: c });
  b(() => {
    d.current = { onFinish: a, onFinishFailed: m, onValuesChange: c };
  }, [a, m, c]), b(() => {
    const e = r;
    typeof e.__bindCallbacks == "function" && e.__bindCallbacks(d.current);
  }, [r, a, m, c]);
  const k = x(void 0);
  b(() => {
    if (!l) return;
    const e = JSON.stringify(l);
    e !== k.current && (k.current = e, r.setFieldsValue(l));
  }, [l, r]);
  const q = I(
    () => ({
      form: r,
      prefixCls: t,
      layout: n,
      labelAlign: y,
      labelCol: C,
      wrapperCol: P,
      size: p,
      disabled: v,
      colon: R,
      requiredMark: g
    }),
    [r, n, y, C, P, p, v, R, g]
  ), w = (e) => {
    e.preventDefault(), e.stopPropagation(), r.validateFields().then(
      (o) => {
        d.current.onFinish?.(o);
      },
      (o) => {
        o && Array.isArray(o.errorFields) && d.current.onFinishFailed?.({
          values: o.values ?? {},
          errorFields: o.errorFields,
          outOfDate: !1
        });
      }
    );
  }, O = (e) => {
    e.preventDefault(), r.resetFields(), $?.(e);
  }, j = J(
    f[t],
    f[`${t}-${n}`],
    f[`${t}-${p}`],
    {
      [f[`${t}-disabled`]]: v
    },
    z
  );
  return /* @__PURE__ */ h(N.Provider, { value: q, children: /* @__PURE__ */ h("form", { ref: F, className: j, onSubmit: w, onReset: O, ...D, children: _ }) });
}
const u = E.forwardRef(B);
u.displayName = "Form";
u.useForm = A;
function G({ form: s, children: F }) {
  const i = I(
    () => ({
      form: s,
      prefixCls: t,
      layout: "vertical",
      labelAlign: "left",
      size: "middle",
      disabled: !1,
      colon: !0,
      requiredMark: !1
    }),
    [s]
  );
  return /* @__PURE__ */ h(N.Provider, { value: i, children: F });
}
const V = G;
V.displayName = "FormProvider";
u.Item = K;
u.Provider = V;
export {
  u as Form
};
