// animal-island-ui/dist/es/components/Form/Form.module.less.js
import "./Form.module.css";
const a = "animal-island-form-fyTI8", i = "animal-island-form-horizontal-JQ-SU", l = "animal-island-form-item-g74aA", n = "animal-island-form-vertical-1-NQN", m = "animal-island-form-item-label-aCJ6a", o = "animal-island-form-inline-Y-Nk0", r = "animal-island-form-small-YW1VY", s = "animal-island-form-middle-FCRU-", e = "animal-island-form-large-UB8sv", d = "animal-island-form-disabled-56MSB", t = "animal-island-form-item-label-required-JQ5sj", f = "animal-island-form-item-label-colon-P-Ted", c = "animal-island-form-item-control-vpLDe", F = "animal-island-form-item-control-input-XM3db", I = "animal-island-form-item-explain-Yk5kW", b = "animal-island-form-item-explain-error--EqdA", p = "animal-island-form-item-feedback-icon-Et6Nr", u = "animal-island-form-item-has-error-E0EuR", g = "animal-island-form-item-has-warning--p8ye", E = "animal-island-form-item-has-success-AJCnO", k = "animal-island-form-item-is-validating-kFRzU", h = {
  "island-form": "animal-island-form-fyTI8",
  islandForm: a,
  "island-form-horizontal": "animal-island-form-horizontal-JQ-SU",
  islandFormHorizontal: i,
  "island-form-item": "animal-island-form-item-g74aA",
  islandFormItem: l,
  "island-form-vertical": "animal-island-form-vertical-1-NQN",
  islandFormVertical: n,
  "island-form-item-label": "animal-island-form-item-label-aCJ6a",
  islandFormItemLabel: m,
  "island-form-inline": "animal-island-form-inline-Y-Nk0",
  islandFormInline: o,
  "island-form-small": "animal-island-form-small-YW1VY",
  islandFormSmall: r,
  "island-form-middle": "animal-island-form-middle-FCRU-",
  islandFormMiddle: s,
  "island-form-large": "animal-island-form-large-UB8sv",
  islandFormLarge: e,
  "island-form-disabled": "animal-island-form-disabled-56MSB",
  islandFormDisabled: d,
  "island-form-item-label-required": "animal-island-form-item-label-required-JQ5sj",
  islandFormItemLabelRequired: t,
  "island-form-item-label-colon": "animal-island-form-item-label-colon-P-Ted",
  islandFormItemLabelColon: f,
  "island-form-item-control": "animal-island-form-item-control-vpLDe",
  islandFormItemControl: c,
  "island-form-item-control-input": "animal-island-form-item-control-input-XM3db",
  islandFormItemControlInput: F,
  "island-form-item-explain": "animal-island-form-item-explain-Yk5kW",
  islandFormItemExplain: I,
  "island-form-item-explain-error": "animal-island-form-item-explain-error--EqdA",
  islandFormItemExplainError: b,
  "island-form-item-feedback-icon": "animal-island-form-item-feedback-icon-Et6Nr",
  islandFormItemFeedbackIcon: p,
  "island-form-item-has-error": "animal-island-form-item-has-error-E0EuR",
  islandFormItemHasError: u,
  "island-form-item-has-warning": "animal-island-form-item-has-warning--p8ye",
  islandFormItemHasWarning: g,
  "island-form-item-has-success": "animal-island-form-item-has-success-AJCnO",
  islandFormItemHasSuccess: E,
  "island-form-item-is-validating": "animal-island-form-item-is-validating-kFRzU",
  islandFormItemIsValidating: k
};
export {
  h as default,
  a as islandForm,
  d as islandFormDisabled,
  i as islandFormHorizontal,
  o as islandFormInline,
  l as islandFormItem,
  c as islandFormItemControl,
  F as islandFormItemControlInput,
  I as islandFormItemExplain,
  b as islandFormItemExplainError,
  p as islandFormItemFeedbackIcon,
  u as islandFormItemHasError,
  E as islandFormItemHasSuccess,
  g as islandFormItemHasWarning,
  k as islandFormItemIsValidating,
  m as islandFormItemLabel,
  f as islandFormItemLabelColon,
  t as islandFormItemLabelRequired,
  e as islandFormLarge,
  s as islandFormMiddle,
  r as islandFormSmall,
  n as islandFormVertical
};
