// animal-island-ui/dist/es/components/Form/useForm.js
import { useRef as R } from "react";
import { runRule as $ } from "./validators.js";
import { stringifyNamePath as f } from "./types.js";
function P() {
  const r = /* @__PURE__ */ new Map();
  return {
    registerField(s, l, c, a) {
      const h = f(s);
      if (!r.has(h))
        r.set(h, {
          key: h,
          rules: l,
          initialValue: c,
          initialSet: c !== void 0,
          currentValue: c,
          notify: a
        });
      else {
        const d = r.get(h);
        d && (d.rules = l, d.notify = a);
      }
    },
    unregisterField(s) {
      r.delete(f(s));
    },
    hasField(s) {
      return r.has(f(s));
    },
    updateRules(s, l) {
      const c = f(s), a = r.get(c);
      a && (a.rules = l);
    },
    notifyAll: () => {
      r.forEach((s) => {
        try {
          s.notify();
        } catch {
        }
      });
    },
    forEachField: (s) => {
      r.forEach((l) => {
        s({ key: l.key, rules: l.rules, notify: l.notify });
      });
    }
  };
}
function m(r, i = "", u = {}) {
  for (const s of Object.keys(r)) {
    const l = i ? `${i}.${s}` : s, c = r[s];
    c !== null && typeof c == "object" && !Array.isArray(c) ? m(c, l, u) : u[l] = c;
  }
  return u;
}
function T(r) {
  const i = /* @__PURE__ */ new Map(), u = /* @__PURE__ */ new Map(), s = /* @__PURE__ */ new Map(), l = /* @__PURE__ */ new Set(), c = /* @__PURE__ */ new Set(), a = P();
  if (r.initialValues) {
    const e = m(r.initialValues);
    Object.keys(e).forEach((n) => {
      i.set(n, e[n]), u.set(n, e[n]);
    });
  }
  function h(e) {
    return i.get(f(e));
  }
  function d(e) {
    const n = {};
    return e === !0 || !e ? i.forEach((t, o) => {
      n[o] = t;
    }) : e.forEach((t) => {
      const o = f(t);
      i.has(o) && (n[o] = i.get(o));
    }), n;
  }
  function b(e, n) {
    const t = f(e), o = i.get(t);
    i.set(t, n), l.add(t), a.forEachField((y) => {
      y.key === t && y.notify();
    }), o !== n && r.onValuesChange?.({ [t]: n }, d(!0));
  }
  function V(e) {
    const n = m(e);
    Object.keys(n).forEach((t) => {
      i.set(t, n[t]), u.has(t) || u.set(t, n[t]);
    }), a.notifyAll();
  }
  function p(e) {
    (e ? e.map(f) : Array.from(i.keys())).forEach((t) => {
      i.set(t, u.get(t)), s.delete(t), l.delete(t);
    }), a.notifyAll();
  }
  function w(e) {
    return l.has(f(e));
  }
  function S(e) {
    return c.has(f(e));
  }
  function A(e) {
    return s.get(f(e));
  }
  function C(e) {
    e.forEach((n) => {
      const t = f(n.name);
      n.value !== void 0 && i.set(t, n.value), n.errors !== void 0 && (n.errors.length === 0 ? s.delete(t) : s.set(t, n.errors)), n.touched && l.add(t), n.validating ? c.add(t) : c.delete(t);
    }), a.notifyAll();
  }
  async function v(e) {
    const n = [];
    a.forEachField((o) => {
      (!e || e.some((y) => f(y) === o.key)) && n.push({ key: o.key, rules: o.rules });
    }), n.forEach((o) => c.add(o.key)), a.notifyAll();
    const t = [];
    if (await Promise.all(
      n.map(async (o) => {
        const y = i.get(o.key), I = o.rules, g = [];
        for (const k of I) {
          const O = typeof k == "function" ? k(F) : k;
          try {
            await $(O, y);
          } catch (E) {
            g.push(E instanceof Error ? E.message : String(E));
          }
        }
        g.length > 0 ? (s.set(o.key, g), l.add(o.key), t.push({ name: o.key, errors: g })) : s.delete(o.key);
      })
    ), n.forEach((o) => c.delete(o.key)), a.notifyAll(), t.length > 0) {
      const o = new Error("Validation failed");
      throw o.errorFields = t, o.values = d(!0), o;
    }
    return d(!0);
  }
  function M() {
    v().then((e) => {
      r.onFinish?.(e);
    }).catch((e) => {
      e && e.errorFields && e.values !== void 0 && r.onFinishFailed?.({
        values: e.values,
        errorFields: e.errorFields,
        outOfDate: !1
      });
    });
  }
  function _(e) {
    e.onFinish && (r.onFinish = e.onFinish), e.onFinishFailed && (r.onFinishFailed = e.onFinishFailed), e.onValuesChange && (r.onValuesChange = e.onValuesChange);
  }
  function j(e, n) {
    const t = f(e);
    typeof document < "u" && document.querySelector(`[data-field-name="${t}"]`)?.scrollIntoView({ behavior: "smooth", block: "center" });
  }
  const F = {
    getFieldValue: h,
    getFieldsValue: d,
    setFieldValue: b,
    setFieldsValue: V,
    resetFields: p,
    validateFields: v,
    submit: M,
    setFields: C,
    isFieldTouched: w,
    isFieldValidating: S,
    getFieldError: A,
    scrollToField: j
  };
  return F.__store = a, F.__bindCallbacks = _, F;
}
function N(r) {
  const i = R(null);
  if (!i.current) {
    const u = T({
      ...r,
      submit: () => u.submit()
    });
    i.current = u;
  }
  return [i.current];
}
export {
  N as useForm
};
