// animal-island-ui/dist/es/components/Select/Select.js
import { jsxs as y, jsx as o } from "react/jsx-runtime";
import { useState as h, useRef as I, useId as M, useEffect as b, useCallback as W } from "react";
import a from "./select.module.less.js";
const O = ({
  options: n,
  value: i,
  onChange: x,
  placeholder: L = "请选择",
  disabled: c = !1,
  "aria-label": k,
  "aria-labelledby": D
}) => {
  const [l, d] = h(!1), [f, s] = h(null), [C, R] = h({}), [B, u] = h(!1), m = I(null), g = I(null), H = n.find((e) => e.key === i)?.label || L, $ = `animal-select-${M().replace(/:/g, "")}`, N = `${$}-listbox`, A = (e) => `${$}-option-${e}`;
  b(() => {
    const e = (t) => {
      m.current && !m.current.contains(t.target) && (d(!1), u(!1));
    };
    return l && document.addEventListener("mousedown", e), () => document.removeEventListener("mousedown", e);
  }, [l]), b(() => {
    if (l && m.current) {
      const e = m.current.getBoundingClientRect(), t = window.innerWidth, w = window.innerHeight, p = n.length * 44 + 24, r = {
        position: "absolute"
      };
      e.right + 200 > t ? (r.right = "100%", r.marginRight = "6px", r.left = "auto") : (r.left = "100%", r.marginLeft = "6px", r.right = "auto");
      const v = w - e.bottom, j = e.top;
      v < p && j > v ? (r.top = "auto", r.bottom = "100%", r.marginBottom = "6px", delete r.transform) : v < p || e.top < p ? (r.top = "100%", r.marginTop = "6px", r.bottom = "auto", delete r.transform) : (r.top = "50%", r.transform = "translateY(-50%)", r.bottom = "auto"), R(r), requestAnimationFrame(() => {
        u(!0);
      });
    } else l || u(!1);
  }, [l, n.length]), b(() => {
    s(l && (i || n[0]?.key) || null);
  }, [l, i, n]);
  const E = W(
    (e) => {
      x(e), d(!1), u(!1), g.current?.focus();
    },
    [x]
  ), S = (e) => {
    if (!n.length) return;
    const t = n.findIndex((p) => p.key === f), w = t < 0 ? e === 1 ? 0 : n.length - 1 : (t + e + n.length) % n.length;
    s(n[w].key);
  }, K = (e) => {
    if (c) return;
    const { key: t } = e;
    if (!l) {
      (t === "Enter" || t === " " || t === "ArrowDown" || t === "ArrowUp") && (e.preventDefault(), d(!0));
      return;
    }
    t === "Escape" ? (e.preventDefault(), d(!1), u(!1), g.current?.focus()) : t === "ArrowDown" ? (e.preventDefault(), S(1)) : t === "ArrowUp" ? (e.preventDefault(), S(-1)) : t === "Home" ? (e.preventDefault(), n[0] && s(n[0].key)) : t === "End" ? (e.preventDefault(), n.length && s(n[n.length - 1].key)) : (t === "Enter" || t === " ") && (e.preventDefault(), f && E(f));
  };
  return /* @__PURE__ */ y(
    "div",
    {
      ref: m,
      className: `${a.wrapper} ${c ? a.disabled : ""}`,
      onKeyDown: K,
      children: [
        /* @__PURE__ */ y(
          "div",
          {
            ref: g,
            className: `${a.trigger} ${l ? a.open : ""}`,
            onClick: () => !c && d(!l),
            role: "combobox",
            "aria-haspopup": "listbox",
            "aria-expanded": l,
            "aria-controls": l ? N : void 0,
            "aria-activedescendant": l && f ? A(f) : void 0,
            "aria-disabled": c || void 0,
            "aria-label": k,
            "aria-labelledby": D,
            tabIndex: c ? -1 : 0,
            children: [
              /* @__PURE__ */ o("span", { className: i ? a.value : a.placeholder, children: H }),
              /* @__PURE__ */ o("span", { className: a.arrow, "aria-hidden": !0, children: /* @__PURE__ */ o("svg", { width: "12", height: "12", viewBox: "0 0 12 12", fill: "none", children: /* @__PURE__ */ o(
                "path",
                {
                  d: "M3 4.5L6 7.5L9 4.5",
                  stroke: "currentColor",
                  strokeWidth: "1.5",
                  strokeLinecap: "round",
                  strokeLinejoin: "round"
                }
              ) }) })
            ]
          }
        ),
        l && B && /* @__PURE__ */ o(
          "div",
          {
            className: a.dropdown,
            style: C,
            role: "listbox",
            id: N,
            "aria-label": k,
            "aria-labelledby": D,
            children: n.map((e) => {
              const t = i === e.key;
              return /* @__PURE__ */ y(
                "div",
                {
                  id: A(e.key),
                  role: "option",
                  "aria-selected": t,
                  className: `${a.option} ${t ? a.active : ""}`,
                  onClick: () => E(e.key),
                  onMouseEnter: () => {
                    s(e.key);
                  },
                  children: [
                    /* @__PURE__ */ o("span", { className: a.optionDot, "aria-hidden": !0 }),
                    e.label,
                    t && /* @__PURE__ */ o("div", { className: a.pillBar })
                  ]
                },
                e.key
              );
            })
          }
        )
      ]
    }
  );
};
O.displayName = "Select";
export {
  O as Select
};
