// animal-island-ui/dist/es/components/Tabs/Tabs.js
import { jsxs as p, jsx as l } from "react/jsx-runtime";
import { useState as j, useId as D, useRef as K, useCallback as v } from "react";
import t from "./tabs.module.less.js";
import E from "../../../files/icon-leaf.40329f03.png";
const H = ({
  items: n,
  defaultActiveKey: I,
  activeKey: c,
  onChange: f,
  className: x,
  style: $,
  leafAnimation: N = !0,
  shadow: g = !0,
  "aria-label": m
}) => {
  const [w, A] = j(I || n[0]?.key), i = c !== void 0 ? c : w, u = `animal-tabs-${D().replace(/:/g, "")}`, y = (e) => `${u}-tab-${e}`, h = (e) => `${u}-panel-${e}`, d = K(/* @__PURE__ */ new Map()), b = v(
    (e) => {
      c === void 0 && A(e), f?.(e);
    },
    [c, f]
  ), L = (e) => {
    d.current.get(e)?.focus();
  }, C = v(
    (e) => {
      const { key: a } = e;
      if (a !== "ArrowRight" && a !== "ArrowLeft" && a !== "Home" && a !== "End")
        return;
      e.preventDefault();
      const r = n.findIndex((T) => T.key === i);
      if (r < 0) return;
      let s = r;
      a === "ArrowRight" ? s = (r + 1) % n.length : a === "ArrowLeft" ? s = (r - 1 + n.length) % n.length : a === "Home" ? s = 0 : a === "End" && (s = n.length - 1);
      const k = n[s].key;
      b(k), L(k);
    },
    [n, i, b]
  ), o = n.find((e) => e.key === i), R = [t.tabs, x].filter(Boolean).join(" ");
  return /* @__PURE__ */ p("div", { className: R, style: $, children: [
    /* @__PURE__ */ l(
      "div",
      {
        className: t.tabList,
        role: "tablist",
        "aria-label": m,
        "aria-orientation": "horizontal",
        onKeyDown: C,
        children: n.map((e) => {
          const a = e.key === i;
          return /* @__PURE__ */ p(
            "button",
            {
              ref: (r) => {
                r ? d.current.set(e.key, r) : d.current.delete(e.key);
              },
              id: y(e.key),
              type: "button",
              role: "tab",
              "aria-selected": a,
              "aria-controls": h(e.key),
              tabIndex: a ? 0 : -1,
              className: `${t.tabItem} ${a ? t.active : ""} ${a && g ? t["active-shadow"] : ""}`,
              onClick: () => b(e.key),
              children: [
                /* @__PURE__ */ l("span", { className: t.tabIcon, "aria-hidden": !0, children: a ? "●" : "○" }),
                /* @__PURE__ */ l("span", { className: t.tabLabel, children: e.label }),
                a && /* @__PURE__ */ l(
                  "img",
                  {
                    src: E,
                    alt: "",
                    className: `${t.tabLeaf} ${N ? "" : t.tabLeafStatic}`
                  }
                )
              ]
            },
            e.key
          );
        })
      }
    ),
    /* @__PURE__ */ l(
      "div",
      {
        className: t.tabContent,
        role: "tabpanel",
        id: o ? h(o.key) : void 0,
        "aria-labelledby": o ? y(o.key) : void 0,
        tabIndex: 0,
        children: /* @__PURE__ */ l("div", { className: t.tabContentInner, children: o?.children })
      }
    )
  ] });
};
H.displayName = "Tabs";
export {
  H as Tabs
};
