// animal-island-ui/dist/es/components/Wallet/Wallet.js
import { jsxs as c, jsx as e } from "react/jsx-runtime";
import i from "./wallet.module.less.js";
import m from "../../../files/item-022.af1e3961.png";
import { Icon as d } from "../Icon/Icon.js";
const f = (r, t) => {
  if (r == null) return "00,000";
  if (typeof r != "number" || !t) return String(r);
  const s = r < 0 ? "-" : "", [a, l] = Math.abs(r).toString().split("."), n = a.replace(/\B(?=(\d{3})+(?!\d))/g, t);
  return l ? `${s}${n}.${l}` : `${s}${n}`;
}, p = {
  small: i["size-small"],
  medium: "",
  large: i["size-large"]
}, g = ({
  value: r = "00,000",
  icon: t,
  size: s = "medium",
  thousandSeparator: a = ",",
  className: l,
  style: n
}) => {
  const o = [i.wallet, p[s], l].filter(Boolean).join(" ");
  return /* @__PURE__ */ c("div", { className: o, style: n, children: [
    t ? /* @__PURE__ */ e("div", { className: i.bagSlot, "aria-hidden": "true", children: t }) : /* @__PURE__ */ e("div", { className: i.bagSlot, "aria-hidden": "true", children: /* @__PURE__ */ e(d, { src: m, size: "80%" }) }),
    /* @__PURE__ */ e("div", { className: i.pill, children: /* @__PURE__ */ e("span", { className: i.value, children: f(r, a) }) })
  ] });
};
g.displayName = "Wallet";
export {
  g as Wallet
};
