// animal-island-ui/dist/es/components/Phone/Phone.js
import { jsx as i, jsxs as c } from "react/jsx-runtime";
import { useState as p, useEffect as N } from "react";
import e from "./phone.module.less.js";
import { Icon as h } from "../Icon/Icon.js";
const f = [
  { id: "camera", iconName: "icon-camera", color: "#B77DEE", hasNewMessage: !0 },
  { id: "app", iconName: "icon-miles", color: "#889DF0", offset: !0 },
  { id: "critterpedia", iconName: "icon-critterpedia", color: "#F7CD67", iconStyle: { width: "105px" } },
  { id: "diy", iconName: "icon-diy", color: "#E59266" },
  { id: "shopping", iconName: "icon-design", color: "#F8A6B2" },
  { id: "variant", iconName: "icon-map", color: "#82D5BB", hasNewMessage: !0, iconStyle: { width: "90px" } },
  { id: "design", iconName: "icon-variant", color: "#8AC68A", iconStyle: { width: "80px" } },
  { id: "map", iconName: "icon-helicopter", color: "#FC736D" },
  { id: "chat", iconName: "icon-chat", color: "#D1DA49" }
], D = ({ className: n }) => {
  const [o, t] = p(/* @__PURE__ */ new Date());
  N(() => {
    const a = setInterval(() => {
      t(/* @__PURE__ */ new Date());
    }, 1e3);
    return () => clearInterval(a);
  }, []);
  const s = o.getHours(), r = o.getMinutes(), l = s >= 12 ? "PM" : "AM", d = s % 12 || 12, m = r.toString().padStart(2, "0");
  return /* @__PURE__ */ i("div", { className: `${e.phoneContainer} ${n || ""}`, children: /* @__PURE__ */ i("div", { className: e.phone, children: /* @__PURE__ */ i("div", { className: e.screenContent, children: /* @__PURE__ */ c("div", { className: e.homeScreen, children: [
    /* @__PURE__ */ c("div", { className: e.dateDisplay, children: [
      /* @__PURE__ */ c("div", { className: e.dateDisplayHeader, children: [
        /* @__PURE__ */ i("span", { className: e.iconWifi }),
        /* @__PURE__ */ c("div", { children: [
          d,
          /* @__PURE__ */ i("span", { className: e.blink, children: ":" }),
          m,
          l
        ] }),
        /* @__PURE__ */ i("span", { className: e.iconLocation })
      ] }),
      /* @__PURE__ */ i("div", { className: e.dayText, children: "Welcome!" })
    ] }),
    /* @__PURE__ */ i("div", { className: e.appsGrid, children: f.map((a) => /* @__PURE__ */ c(
      "div",
      {
        className: `${e.appItem} ${a.offset ? e.appItemOffset : ""}`,
        style: { backgroundColor: a.color },
        children: [
          a.hasNewMessage && /* @__PURE__ */ i("span", { className: e.badge }),
          /* @__PURE__ */ i(
            h,
            {
              name: a.iconName,
              size: "100%",
              className: `${e.appIcon} ${a.offset ? e.appIconOffset : ""}`,
              style: { backgroundSize: "70% auto", ...a.iconStyle }
            }
          )
        ]
      },
      a.id
    )) }),
    /* @__PURE__ */ i("div", { className: e.pageIndicator, children: /* @__PURE__ */ i("span", { className: e.iconPage }) })
  ] }) }) }) });
};
export {
  D as Phone
};
