// animal-island-ui/dist/es/components/Phone/phone.module.less.js
import "./phone.module.css";
const a = "animal-phoneContainer-49yKw", n = "animal-phone-bXgSa", e = "animal-screenContent-oyJF-", o = "animal-homeScreen-no2yS", t = "animal-dateDisplay-cnvu8", i = "animal-dateDisplayHeader-32i0J", c = "animal-blink--R0wf", p = "animal-dayText-7NEMb", s = "animal-appsGrid-opTs8", m = "animal-appItem-1NmDt", l = "animal-appIcon-tTPIg", d = "animal-appIconOffset-b5SZG", r = "animal-appItemOffset-FEStg", f = "animal-iconWifi-RoLJH", I = "animal-iconLocation-YfhoK", g = "animal-iconPage-wMmBC", y = "animal-badge-drGmk", h = "animal-pageIndicator-mQ-Pg", b = {
  phoneContainer: a,
  phone: n,
  screenContent: e,
  homeScreen: o,
  dateDisplay: t,
  dateDisplayHeader: i,
  blink: c,
  dayText: p,
  appsGrid: s,
  appItem: m,
  appIcon: l,
  appIconOffset: d,
  appItemOffset: r,
  iconWifi: f,
  iconLocation: I,
  iconPage: g,
  badge: y,
  pageIndicator: h
};
export {
  l as appIcon,
  d as appIconOffset,
  m as appItem,
  r as appItemOffset,
  s as appsGrid,
  y as badge,
  c as blink,
  t as dateDisplay,
  i as dateDisplayHeader,
  p as dayText,
  b as default,
  o as homeScreen,
  I as iconLocation,
  g as iconPage,
  f as iconWifi,
  h as pageIndicator,
  n as phone,
  a as phoneContainer,
  e as screenContent
};
