// animal-island-ui/dist/es/components/Collapse/Collapse.js
import { jsxs as o, jsx as a } from "react/jsx-runtime";
import { useState as u, useId as x } from "react";
import e from "./collapse.module.less.js";
const N = ({
  question: d,
  answer: t,
  defaultExpanded: c = !1,
  disabled: s = !1,
  className: p,
  style: h
}) => {
  const [n, m] = u(c), l = `animal-collapse-${x().replace(/:/g, "")}`, r = `${l}-header`, i = `${l}-panel`, C = () => {
    s || m(!n);
  }, f = [e.faqCard, n && e.expanded, s && e.disabled, p].filter(Boolean).join(" ");
  return /* @__PURE__ */ o("div", { className: f, style: h, children: [
    /* @__PURE__ */ o(
      "button",
      {
        type: "button",
        id: r,
        className: e.questionHeader,
        onClick: C,
        disabled: s,
        "aria-expanded": n,
        "aria-controls": i,
        children: [
          /* @__PURE__ */ a("span", { className: e.questionIcon, "aria-hidden": !0, children: n ? "−" : "+" }),
          /* @__PURE__ */ a("span", { className: e.questionText, children: d }),
          /* @__PURE__ */ a("span", { className: e.leafDecoration, "aria-hidden": !0, children: /* @__PURE__ */ a("svg", { viewBox: "0 0 24 24", width: "20", height: "20", children: /* @__PURE__ */ a(
            "path",
            {
              fill: "currentColor",
              d: "M17,8C8,10 5.9,16.17 3.82,21.34L5.71,22L6.66,19.7C7.14,19.87 7.64,20 8,20C19,20 22,3 22,3C21,5 14,5.25 9,6.25C4,7.25 2,11.5 2,13.5C2,15.5 3.75,17.25 3.75,17.25C7,8 17,8 17,8Z"
            }
          ) }) })
        ]
      }
    ),
    /* @__PURE__ */ a("div", { className: e.answerWrapper, id: i, role: "region", "aria-labelledby": r, children: /* @__PURE__ */ a("div", { className: e.answerContent, children: t }) })
  ] });
};
N.displayName = "Collapse";
export {
  N as Collapse
};
