// animal-island-ui/dist/es/components/Input/Input.js
import { jsxs as E, jsx as a } from "react/jsx-runtime";
import { useState as S, useRef as k, useCallback as d } from "react";
import e from "./input.module.less.js";
const O = ({
  size: w = "middle",
  prefix: s,
  suffix: l,
  allowClear: h = !1,
  status: p,
  shadow: v = !1,
  disabled: r = !1,
  className: b,
  value: o,
  defaultValue: I,
  onChange: c,
  onClear: i,
  clearAriaLabel: N = "清除",
  ...y
}) => {
  const [C, u] = S(I ?? ""), n = o !== void 0, m = n ? o : C, f = k(null), V = d(
    (t) => {
      n || u(t.target.value), c?.(t);
    },
    [n, c]
  ), j = d(() => {
    n || u(""), i?.();
    const t = f.current;
    if (t) {
      Object.getOwnPropertyDescriptor(
        window.HTMLInputElement.prototype,
        "value"
      )?.set?.call(t, "");
      const x = new Event("input", { bubbles: !0 });
      t.dispatchEvent(x);
    }
  }, [n, i]), g = [
    e.wrapper,
    e[`wrapper-${w}`],
    p && e[`wrapper-${p}`],
    r && e["wrapper-disabled"],
    !v && e["wrapper-no-shadow"],
    b
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ E("span", { className: g, children: [
    s && /* @__PURE__ */ a("span", { className: e.prefix, children: s }),
    /* @__PURE__ */ a(
      "input",
      {
        ref: f,
        className: e.input,
        disabled: r,
        value: m,
        onChange: V,
        ...y
      }
    ),
    h && m && !r && /* @__PURE__ */ a("button", { type: "button", className: e.clear, onClick: j, "aria-label": N, children: "×" }),
    l && /* @__PURE__ */ a("span", { className: e.suffix, children: l })
  ] });
};
O.displayName = "Input";
export {
  O as Input
};
