// animal-island-ui/dist/es/components/Input/input.module.less.js
import "./input.module.css";
const r = "animal-wrapper-IRrhj", a = "animal-wrapper-disabled-MWwHK", p = "animal-input-12WUn", e = "animal-wrapper-no-shadow-WP9Gk", n = "animal-wrapper-small-mWvja", w = "animal-wrapper-middle-zXqMR", l = "animal-wrapper-large-RrYM4", i = "animal-wrapper-error-W-VRo", o = "animal-wrapper-warning-miO7A", m = "animal-prefix-g8f1R", s = "animal-suffix-o7ywP", d = "animal-clear-HuR2N", t = {
  wrapper: r,
  "wrapper-disabled": "animal-wrapper-disabled-MWwHK",
  wrapperDisabled: a,
  input: p,
  "wrapper-no-shadow": "animal-wrapper-no-shadow-WP9Gk",
  wrapperNoShadow: e,
  "wrapper-small": "animal-wrapper-small-mWvja",
  wrapperSmall: n,
  "wrapper-middle": "animal-wrapper-middle-zXqMR",
  wrapperMiddle: w,
  "wrapper-large": "animal-wrapper-large-RrYM4",
  wrapperLarge: l,
  "wrapper-error": "animal-wrapper-error-W-VRo",
  wrapperError: i,
  "wrapper-warning": "animal-wrapper-warning-miO7A",
  wrapperWarning: o,
  prefix: m,
  suffix: s,
  clear: d
};
export {
  d as clear,
  t as default,
  p as input,
  m as prefix,
  s as suffix,
  r as wrapper,
  a as wrapperDisabled,
  i as wrapperError,
  l as wrapperLarge,
  w as wrapperMiddle,
  e as wrapperNoShadow,
  n as wrapperSmall,
  o as wrapperWarning
};
