// animal-island-ui/dist/es/components/Button/button.module.less.js
import "./button.module.css";
const n = "animal-btn-QoKe1", t = "animal-btn-small-6Srsd", a = "animal-btn-middle-zEXqv", b = "animal-btn-large-k9IvA", l = "animal-btn-default-L7yE6", i = "animal-btn-primary-9tMG9", m = "animal-btn-dashed-wzZ-i", o = "animal-btn-text-Tope4", d = "animal-btn-link-pURnf", e = "animal-btn-danger-zQ4kF", s = "animal-btn-ghost-Ykty4", c = "animal-btn-block-MC4pp", g = "animal-btn-loading-PfXLM", r = "animal-animal-btn-loading-Rf1hg", k = "animal-btn-icon-P5CS9", f = {
  btn: n,
  "btn-small": "animal-btn-small-6Srsd",
  btnSmall: t,
  "btn-middle": "animal-btn-middle-zEXqv",
  btnMiddle: a,
  "btn-large": "animal-btn-large-k9IvA",
  btnLarge: b,
  "btn-default": "animal-btn-default-L7yE6",
  btnDefault: l,
  "btn-primary": "animal-btn-primary-9tMG9",
  btnPrimary: i,
  "btn-dashed": "animal-btn-dashed-wzZ-i",
  btnDashed: m,
  "btn-text": "animal-btn-text-Tope4",
  btnText: o,
  "btn-link": "animal-btn-link-pURnf",
  btnLink: d,
  "btn-danger": "animal-btn-danger-zQ4kF",
  btnDanger: e,
  "btn-ghost": "animal-btn-ghost-Ykty4",
  btnGhost: s,
  "btn-block": "animal-btn-block-MC4pp",
  btnBlock: c,
  "btn-loading": "animal-btn-loading-PfXLM",
  btnLoading: g,
  "animal-btn-loading": "animal-animal-btn-loading-Rf1hg",
  animalBtnLoading: r,
  "btn-icon": "animal-btn-icon-P5CS9",
  btnIcon: k
};
export {
  r as animalBtnLoading,
  n as btn,
  c as btnBlock,
  e as btnDanger,
  m as btnDashed,
  l as btnDefault,
  s as btnGhost,
  k as btnIcon,
  b as btnLarge,
  d as btnLink,
  g as btnLoading,
  a as btnMiddle,
  i as btnPrimary,
  t as btnSmall,
  o as btnText,
  f as default
};
