// animal-island-ui/dist/es/components/Radio/radio.module.less.js
import "./radio.module.css";
const a = "animal-radioGroup-J1a5S", l = "animal-horizontal-DOVJ9", i = "animal-vertical-fBz4U", s = "animal-radioItem-nzJp0", n = "animal-small-bKLJe", o = "animal-label-4sFiA", c = "animal-middle-F-4OZ", e = "animal-large-cB3ul", d = "animal-cbx-4s--B", t = "animal-splash-gCU-H", m = "animal-check--tFZL", r = "animal-animal-radio-splash-Mj7UU", h = "animal-checked-t-CwE", p = "animal-disabled-Uthn7", b = "animal-groupDisabled-DfMed", u = {
  radioGroup: a,
  horizontal: l,
  vertical: i,
  radioItem: s,
  small: n,
  label: o,
  middle: c,
  large: e,
  cbx: d,
  splash: t,
  check: m,
  "animal-radio-splash": "animal-animal-radio-splash-Mj7UU",
  animalRadioSplash: r,
  checked: h,
  disabled: p,
  groupDisabled: b
};
export {
  r as animalRadioSplash,
  d as cbx,
  m as check,
  h as checked,
  u as default,
  p as disabled,
  b as groupDisabled,
  l as horizontal,
  o as label,
  e as large,
  c as middle,
  a as radioGroup,
  s as radioItem,
  n as small,
  t as splash,
  i as vertical
};
