// animal-island-ui/dist/es/components/Radio/Radio.js
import { jsx as l, jsxs as x } from "react/jsx-runtime";
import { useState as I, useId as K, useRef as M, useEffect as P, useMemo as k, useCallback as w } from "react";
import n from "./radio.module.less.js";
import D from "classnames";
const S = ({
  value: b,
  defaultValue: N,
  options: s,
  size: y = "middle",
  disabled: c = !1,
  direction: R = "horizontal",
  onChange: p,
  className: A,
  style: C
}) => {
  const [E, F] = I(N), o = b !== void 0, d = o ? b : E, g = `animal-radio-${K().replace(/:/g, "")}`, v = M([]), [u, f] = I(() => {
    const e = s.findIndex((a) => a.value === d);
    return e >= 0 ? e : 0;
  });
  P(() => {
    const e = s.findIndex((a) => a.value === d);
    e >= 0 && f(e);
  }, [d]);
  const r = k(() => s.map((e, a) => ({ opt: e, idx: a })).filter(({ opt: e }) => !c && !e.disabled).map(({ idx: e }) => e), [s, c]), h = k(() => r.indexOf(u), [r, u]), m = w(
    (e, a) => {
      c || a || (o || F(e), p?.(e));
    },
    [c, o, p]
  ), L = w(
    (e) => {
      if (r.length === 0) return;
      let a = -1;
      switch (e.key) {
        case "ArrowRight":
        case "ArrowDown":
          e.preventDefault(), a = (h + 1) % r.length;
          break;
        case "ArrowLeft":
        case "ArrowUp":
          e.preventDefault(), a = (h - 1 + r.length) % r.length;
          break;
        case "Home":
          e.preventDefault(), a = 0;
          break;
        case "End":
          e.preventDefault(), a = r.length - 1;
          break;
        default:
          return;
      }
      if (a >= 0) {
        const t = r[a];
        f(t), m(s[t].value, s[t].disabled), v.current[t]?.focus();
      }
    },
    [r, h, s, m]
  );
  return /* @__PURE__ */ l(
    "div",
    {
      className: D(
        n.radioGroup,
        n[R],
        { [n.groupDisabled]: c },
        A
      ),
      style: C,
      role: "radiogroup",
      onKeyDown: L,
      children: s.map((e, a) => {
        const t = d === e.value, i = c || e.disabled, $ = a === u && !i, j = `${g}-${a}`;
        return /* @__PURE__ */ x(
          "label",
          {
            className: D(n.radioItem, n[y], {
              [n.checked]: t,
              [n.disabled]: i
            }),
            children: [
              /* @__PURE__ */ x("span", { className: n.cbx, children: [
                /* @__PURE__ */ l(
                  "input",
                  {
                    id: j,
                    ref: (B) => {
                      v.current[a] = B;
                    },
                    type: "radio",
                    name: g,
                    checked: t,
                    disabled: i,
                    tabIndex: $ ? 0 : -1,
                    onChange: () => m(e.value, e.disabled),
                    onFocus: () => {
                      i || f(a);
                    }
                  }
                ),
                /* @__PURE__ */ l("span", { className: n.splash, "aria-hidden": "true" }),
                /* @__PURE__ */ l("svg", { className: n.check, fill: "none", viewBox: "0 0 15 14", height: 14, width: 15, children: /* @__PURE__ */ l("path", { d: "M2 8.36364L6.23077 12L13 2" }) })
              ] }),
              /* @__PURE__ */ l("span", { className: n.label, children: e.label })
            ]
          },
          String(e.value)
        );
      })
    }
  );
};
S.displayName = "Radio";
export {
  S as Radio
};
