// animal-island-ui/dist/es/components/Title/title.module.less.js
import "./title.module.css";
const o = "animal-title-3j08r", l = "animal-ribbon-S4CL0", r = "animal-ribbonText-FUXBT", n = "animal-ribbonBack-cW93o", a = "animal-ribbonBackLeft-TZiON", c = "animal-ribbonBackRight-AipiN", p = "animal-ribbonFold-qTBSV", i = "animal-ribbonFoldLeft-gVuCs", e = "animal-ribbonFoldRight-94--n", b = "animal-ribbonFront-qihIp", t = "animal-color-app-pink-m6nip", m = "animal-color-purple-atkjz", s = "animal-color-app-blue-Y3iTY", k = "animal-color-app-yellow-JDjDu", g = "animal-color-app-orange-NmeMm", B = "animal-color-app-teal-QdItW", d = "animal-color-app-green-E-Bfq", w = "animal-color-app-red-Bn9jO", A = "animal-color-lime-green-TD3ck", u = "animal-color-yellow-green-KnmJ5", F = "animal-color-brown-Xr3XB", h = "animal-color-warm-peach-pink-4dp6n", T = {
  title: o,
  ribbon: l,
  ribbonText: r,
  ribbonBack: n,
  ribbonBackLeft: a,
  ribbonBackRight: c,
  ribbonFold: p,
  ribbonFoldLeft: i,
  ribbonFoldRight: e,
  ribbonFront: b,
  "color-app-pink": "animal-color-app-pink-m6nip",
  colorAppPink: t,
  "color-purple": "animal-color-purple-atkjz",
  colorPurple: m,
  "color-app-blue": "animal-color-app-blue-Y3iTY",
  colorAppBlue: s,
  "color-app-yellow": "animal-color-app-yellow-JDjDu",
  colorAppYellow: k,
  "color-app-orange": "animal-color-app-orange-NmeMm",
  colorAppOrange: g,
  "color-app-teal": "animal-color-app-teal-QdItW",
  colorAppTeal: B,
  "color-app-green": "animal-color-app-green-E-Bfq",
  colorAppGreen: d,
  "color-app-red": "animal-color-app-red-Bn9jO",
  colorAppRed: w,
  "color-lime-green": "animal-color-lime-green-TD3ck",
  colorLimeGreen: A,
  "color-yellow-green": "animal-color-yellow-green-KnmJ5",
  colorYellowGreen: u,
  "color-brown": "animal-color-brown-Xr3XB",
  colorBrown: F,
  "color-warm-peach-pink": "animal-color-warm-peach-pink-4dp6n",
  colorWarmPeachPink: h
};
export {
  s as colorAppBlue,
  d as colorAppGreen,
  g as colorAppOrange,
  t as colorAppPink,
  w as colorAppRed,
  B as colorAppTeal,
  k as colorAppYellow,
  F as colorBrown,
  A as colorLimeGreen,
  m as colorPurple,
  h as colorWarmPeachPink,
  u as colorYellowGreen,
  T as default,
  l as ribbon,
  n as ribbonBack,
  a as ribbonBackLeft,
  c as ribbonBackRight,
  p as ribbonFold,
  i as ribbonFoldLeft,
  e as ribbonFoldRight,
  b as ribbonFront,
  r as ribbonText,
  o as title
};
