// animal-island-ui/dist/es/components/Title/Title.js
import { jsx as e, jsxs as o } from "react/jsx-runtime";
import s from "classnames";
import a from "./title.module.less.js";
const b = {
  small: 14,
  middle: 20,
  large: 28
}, d = ({
  children: n,
  fontSize: r,
  color: i
}) => /* @__PURE__ */ o(
  "span",
  {
    className: s(a.ribbon, i && i !== "default" && a[`color-${i}`]),
    style: { fontSize: `${r}px` },
    children: [
      /* @__PURE__ */ e("span", { className: s(a.ribbonBack, a.ribbonBackLeft), "aria-hidden": !0 }),
      /* @__PURE__ */ e("span", { className: s(a.ribbonBack, a.ribbonBackRight), "aria-hidden": !0 }),
      /* @__PURE__ */ e("span", { className: s(a.ribbonFold, a.ribbonFoldLeft), "aria-hidden": !0 }),
      /* @__PURE__ */ e("span", { className: s(a.ribbonFold, a.ribbonFoldRight), "aria-hidden": !0 }),
      /* @__PURE__ */ e("span", { className: a.ribbonFront, "aria-hidden": !0 }),
      /* @__PURE__ */ e("span", { className: a.ribbonText, children: n })
    ]
  }
), m = ({ children: n, size: r = "middle", color: i = "default", className: t, style: l }) => /* @__PURE__ */ e("span", { className: s(a.title, t), style: l, children: /* @__PURE__ */ e(d, { fontSize: b[r], color: i, children: n }) });
m.displayName = "Title";
export {
  m as Title
};
