// animal-island-ui/dist/es/components/Checkbox/checkbox.module.less.js
import "./checkbox.module.css";
const a = "animal-checkboxGroup-q3P6l", l = "animal-horizontal-fBBSw", c = "animal-vertical-klL45", s = "animal-checkboxItem-DxKFM", e = "animal-small-2wCHI", n = "animal-label-L0ila", i = "animal-middle-VUB-E", o = "animal-large-QQxPe", m = "animal-cbx-r-W7r", t = "animal-splash-95w6m", b = "animal-check-aNoPL", h = "animal-animal-cbx-splash-dEVmu", d = "animal-checked-S7gPe", r = "animal-disabled-zDC-1", x = "animal-groupDisabled-LvBFd", p = {
  checkboxGroup: a,
  horizontal: l,
  vertical: c,
  checkboxItem: s,
  small: e,
  label: n,
  middle: i,
  large: o,
  cbx: m,
  splash: t,
  check: b,
  "animal-cbx-splash": "animal-animal-cbx-splash-dEVmu",
  animalCbxSplash: h,
  checked: d,
  disabled: r,
  groupDisabled: x
};
export {
  h as animalCbxSplash,
  m as cbx,
  b as check,
  a as checkboxGroup,
  s as checkboxItem,
  d as checked,
  p as default,
  r as disabled,
  x as groupDisabled,
  l as horizontal,
  n as label,
  o as large,
  i as middle,
  e as small,
  t as splash,
  c as vertical
};
