// animal-island-ui/dist/es/components/Checkbox/Checkbox.js
import { jsx as c, jsxs as h } from "react/jsx-runtime";
import { useState as y, useId as $, useCallback as j } from "react";
import e from "./checkbox.module.less.js";
import m from "classnames";
const w = ({
  value: t,
  defaultValue: u = [],
  options: b,
  size: p = "middle",
  disabled: n = !1,
  direction: x = "horizontal",
  onChange: o,
  className: k,
  style: f
}) => {
  const [N, g] = y(u), r = t !== void 0, a = r ? t : N, v = `animal-cbx-${$().replace(/:/g, "")}`, C = j(
    (s, d) => {
      if (n || d) return;
      const l = a.includes(s) ? a.filter((i) => i !== s) : [...a, s];
      r || g(l), o?.(l);
    },
    [n, a, r, o]
  );
  return /* @__PURE__ */ c(
    "div",
    {
      role: "group",
      className: m(
        e.checkboxGroup,
        e[x],
        { [e.groupDisabled]: n },
        k
      ),
      style: f,
      children: b.map((s, d) => {
        const l = a.includes(s.value), i = n || s.disabled, I = `${v}-${d}`;
        return /* @__PURE__ */ h(
          "label",
          {
            className: m(e.checkboxItem, e[p], {
              [e.checked]: l,
              [e.disabled]: i
            }),
            children: [
              /* @__PURE__ */ h("span", { className: e.cbx, children: [
                /* @__PURE__ */ c(
                  "input",
                  {
                    id: I,
                    type: "checkbox",
                    checked: l,
                    disabled: i,
                    onChange: () => C(s.value, s.disabled)
                  }
                ),
                /* @__PURE__ */ c("span", { className: e.splash, "aria-hidden": "true" }),
                /* @__PURE__ */ c("svg", { className: e.check, fill: "none", viewBox: "0 0 15 14", height: 14, width: 15, children: /* @__PURE__ */ c("path", { d: "M2 8.36364L6.23077 12L13 2" }) })
              ] }),
              /* @__PURE__ */ c("span", { className: e.label, children: s.label })
            ]
          },
          String(s.value)
        );
      })
    }
  );
};
w.displayName = "Checkbox";
export {
  w as Checkbox
};
