// animal-island-ui/dist/es/components/Switch/Switch.js
import { jsxs as x, jsx as i } from "react/jsx-runtime";
import { useState as j, useCallback as h } from "react";
import e from "./switch.module.less.js";
const v = ({
  checked: r,
  defaultChecked: f = !1,
  size: m = "default",
  disabled: n = !1,
  loading: t = !1,
  checkedChildren: p,
  unCheckedChildren: d,
  onChange: o,
  className: w,
  "aria-label": y,
  "aria-labelledby": k
}) => {
  const [u, b] = j(f), c = r !== void 0, a = c ? r : u, l = h(() => {
    if (n || t) return;
    const s = !a;
    c || b(s), o?.(s);
  }, [n, t, a, c, o]), C = h(
    (s) => {
      (s.key === " " || s.key === "Enter") && (s.preventDefault(), l());
    },
    [l]
  ), N = [
    e.switch,
    e[`switch-${m}`],
    a && e["switch-checked"],
    n && e["switch-disabled"],
    t && e["switch-loading"],
    w
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ x(
    "button",
    {
      type: "button",
      role: "switch",
      "aria-checked": a,
      "aria-label": y,
      "aria-labelledby": k,
      "aria-busy": t || void 0,
      className: N,
      onClick: l,
      onKeyDown: C,
      disabled: n,
      children: [
        /* @__PURE__ */ i("span", { className: e.handle, children: t && /* @__PURE__ */ i("span", { className: e.spinner }) }),
        /* @__PURE__ */ i("span", { className: e.inner, children: a ? p : d })
      ]
    }
  );
};
v.displayName = "Switch";
export {
  v as Switch
};
