// animal-island-ui/dist/es/components/Tooltip/tooltip.module.less.js
import "./tooltip.module.css";
const t = "animal-tooltipWrapper-PL7BH", n = "animal-tooltip-UY7BU", o = "animal-visible-Gdi6E", a = "animal-content-GvqM2", l = "animal-island-xz5YK", i = "animal-bordered-FDN3c", s = "animal-borderless-LWkMH", r = "animal-islandBody-HX1M-", e = "animal-islandSvg-VCYbI", m = "animal-islandContent-hg3kA", d = "animal-tail-BKxCA", c = "animal-top-j1fp9", _ = "animal-top_start-gGZfo", p = "animal-top_start-gGZfo", b = "animal-top_end-oSOuZ", g = "animal-top_end-oSOuZ", f = "animal-bottom-1EwHs", h = "animal-bottom_start-CBvpW", S = "animal-bottom_start-CBvpW", E = "animal-bottom_end-J080W", v = "animal-bottom_end-J080W", B = "animal-left-lmx5f", W = "animal-left_start-ror3m", C = "animal-left_start-ror3m", G = "animal-left_end-76zmA", x = "animal-left_end-76zmA", y = "animal-right--IMoa", A = "animal-right_start--T033", H = "animal-right_start--T033", J = "animal-right_end-nGJh5", M = "animal-right_end-nGJh5", Z = {
  tooltipWrapper: t,
  tooltip: n,
  visible: o,
  content: a,
  island: l,
  bordered: i,
  borderless: s,
  islandBody: r,
  islandSvg: e,
  islandContent: m,
  tail: d,
  top: c,
  top_start: _,
  topStart: p,
  top_end: b,
  topEnd: g,
  bottom: f,
  bottom_start: h,
  bottomStart: S,
  bottom_end: E,
  bottomEnd: v,
  left: B,
  left_start: W,
  leftStart: C,
  left_end: G,
  leftEnd: x,
  right: y,
  right_start: A,
  rightStart: H,
  right_end: J,
  rightEnd: M
};
export {
  i as bordered,
  s as borderless,
  f as bottom,
  v as bottomEnd,
  S as bottomStart,
  E as bottom_end,
  h as bottom_start,
  a as content,
  Z as default,
  l as island,
  r as islandBody,
  m as islandContent,
  e as islandSvg,
  B as left,
  x as leftEnd,
  C as leftStart,
  G as left_end,
  W as left_start,
  y as right,
  M as rightEnd,
  H as rightStart,
  J as right_end,
  A as right_start,
  d as tail,
  n as tooltip,
  t as tooltipWrapper,
  c as top,
  g as topEnd,
  p as topStart,
  b as top_end,
  _ as top_start,
  o as visible
};
