// animal-island-ui/dist/es/components/Tooltip/Tooltip.js
import { jsxs as u, jsx as o, Fragment as g } from "react/jsx-runtime";
import b, { useState as j, useRef as x, useId as A, useCallback as N, useEffect as R } from "react";
import e from "./tooltip.module.less.js";
import I from "classnames";
const y = "M0.501,0.005 L0.501,0.005 L0.523,0.005 L0.549,0.006 C0.704,0.01,0.796,0.017,0.825,0.027 L0.827,0.028 C0.872,0.045,0.939,0.044,0.978,0.17 C1,0.254,1,0.365,0.99,0.505 L0.988,0.513 C0.979,0.558,0.971,0.598,0.965,0.633 C0.956,0.689,0.979,0.77,0.964,0.865 C0.953,0.928,0.921,0.966,0.869,0.979 C0.821,0.986,0.773,0.992,0.726,0.995 L0.712,0.996 L0.694,0.997 C0.648,1,0.586,1,0.507,1 L0.501,1 L0.464,1 C0.385,1,0.325,0.998,0.283,0.995 C0.234,0.992,0.184,0.987,0.133,0.979 C0.081,0.966,0.05,0.928,0.039,0.865 C0.023,0.77,0.047,0.689,0.037,0.633 C0.031,0.595,0.023,0.552,0.013,0.505 C-0.006,0.365,-0.002,0.254,0.024,0.17 C0.064,0.045,0.13,0.045,0.174,0.028 L0.175,0.028 C0.204,0.017,0.303,0.009,0.474,0.005 L0.501,0.005", _ = "rgb(247, 243, 223)", w = "#c4b89e", D = ({ id: l }) => /* @__PURE__ */ o("svg", { style: { position: "absolute", width: 0, height: 0 }, "aria-hidden": !0, children: /* @__PURE__ */ o("clipPath", { id: l, clipPathUnits: "objectBoundingBox", children: /* @__PURE__ */ o("path", { d: y }) }) }), F = () => /* @__PURE__ */ o("svg", { className: e.islandSvg, viewBox: "0 0 1 1", preserveAspectRatio: "none", "aria-hidden": !0, children: /* @__PURE__ */ o(
  "path",
  {
    d: y,
    fill: _,
    stroke: w,
    strokeWidth: 2,
    vectorEffect: "non-scaling-stroke",
    strokeLinejoin: "round"
  }
) }), $ = ({
  title: l,
  placement: k = "top",
  trigger: t = "hover",
  variant: B = "default",
  bordered: m = !0,
  children: S,
  className: E,
  style: M
}) => {
  const [c, r] = j(!1), d = x(), v = A().replace(/:/g, ""), C = `animal-tooltip-clip-${v}`, f = `animal-tooltip-${v}`, p = N(() => {
    clearTimeout(d.current), r(!0);
  }, []), h = N(() => {
    d.current = setTimeout(() => r(!1), 100);
  }, []);
  R(() => () => clearTimeout(d.current), []);
  const a = b.Children.only(S), n = a.props, i = {
    // 让屏幕阅读器在 trigger 获取焦点 / hover 描述时能读出 tooltip 内容
    // 仅在显示时生效，避免隐藏 tooltip 仍被读到
    "aria-describedby": c ? [a.props["aria-describedby"], f].filter(Boolean).join(" ") : a.props["aria-describedby"]
  };
  t === "hover" ? (i.onMouseEnter = (s) => {
    p(), n.onMouseEnter?.(s);
  }, i.onMouseLeave = (s) => {
    h(), n.onMouseLeave?.(s);
  }) : t === "focus" ? (i.onFocus = (s) => {
    p(), n.onFocus?.(s);
  }, i.onBlur = (s) => {
    h(), n.onBlur?.(s);
  }) : t === "click" && (i.onClick = (s) => {
    r((T) => !T), n.onClick?.(s);
  });
  const P = e[k.replace(/-/g, "_")], L = B === "island";
  return /* @__PURE__ */ u("div", { className: I(e.tooltipWrapper, E), style: M, children: [
    b.cloneElement(a, i),
    /* @__PURE__ */ o(
      "div",
      {
        className: I(
          e.tooltip,
          P,
          L && e.island,
          m ? e.bordered : e.borderless,
          { [e.visible]: c }
        ),
        role: "tooltip",
        id: f,
        "aria-hidden": !c,
        onMouseEnter: t === "hover" ? p : void 0,
        onMouseLeave: t === "hover" ? h : void 0,
        children: L ? /* @__PURE__ */ u(g, { children: [
          /* @__PURE__ */ u("div", { className: e.islandBody, children: [
            /* @__PURE__ */ o(D, { id: C }),
            m && /* @__PURE__ */ o(F, {}),
            /* @__PURE__ */ o("div", { className: e.islandContent, style: { clipPath: `url(#${C})` }, children: /* @__PURE__ */ o("div", { className: e.content, children: l }) })
          ] }),
          /* @__PURE__ */ o("span", { className: e.tail, "aria-hidden": !0 })
        ] }) : /* @__PURE__ */ o("div", { className: e.content, children: l })
      }
    )
  ] });
};
$.displayName = "Tooltip";
export {
  $ as Tooltip
};
