// animal-island-ui/dist/es/components/Modal/Modal.js
import { jsx as e, jsxs as m, Fragment as K } from "react/jsx-runtime";
import { useState as $, useEffect as c, useRef as k, useCallback as w, useId as O } from "react";
import { createPortal as U } from "react-dom";
import o from "./modal.module.less.js";
import { Cursor as q } from "../Cursor/Cursor.js";
import { Typewriter as _ } from "../Typewriter/Typewriter.js";
import { Button as x } from "../Button/Button.js";
const z = [
  "a[href]",
  "area[href]",
  "button:not([disabled])",
  'input:not([disabled]):not([type="hidden"])',
  "select:not([disabled])",
  "textarea:not([disabled])",
  '[tabindex]:not([tabindex="-1"])',
  "audio[controls]",
  "video[controls]",
  '[contenteditable]:not([contenteditable="false"])'
].join(","), E = (t) => Array.from(t.querySelectorAll(z)).filter(
  (n) => !n.hasAttribute("disabled") && n.getAttribute("aria-hidden") !== "true"
), G = () => /* @__PURE__ */ e("svg", { style: { position: "absolute", width: 0, height: 0 }, "aria-hidden": !0, children: /* @__PURE__ */ e("clipPath", { id: "animal-modal-clip", clipPathUnits: "objectBoundingBox", children: /* @__PURE__ */ e("path", { d: "M0.501,0.005 L0.501,0.005 L0.523,0.005 L0.549,0.006 C0.704,0.01,0.796,0.017,0.825,0.027 L0.827,0.028 C0.872,0.045,0.939,0.044,0.978,0.17 C1,0.254,1,0.365,0.99,0.505 L0.988,0.513 C0.979,0.558,0.971,0.598,0.965,0.633 C0.956,0.689,0.979,0.77,0.964,0.865 C0.953,0.928,0.921,0.966,0.869,0.979 C0.821,0.986,0.773,0.992,0.726,0.995 L0.712,0.996 L0.694,0.997 C0.648,1,0.586,1,0.507,1 L0.501,1 L0.464,1 C0.385,1,0.325,0.998,0.283,0.995 C0.234,0.992,0.184,0.987,0.133,0.979 C0.081,0.966,0.05,0.928,0.039,0.865 C0.023,0.77,0.047,0.689,0.037,0.633 C0.031,0.595,0.023,0.552,0.013,0.505 C-0.006,0.365,-0.002,0.254,0.024,0.17 C0.064,0.045,0.13,0.045,0.174,0.028 L0.175,0.028 C0.204,0.017,0.303,0.009,0.474,0.005 L0.501,0.005" }) }) }), H = ({
  open: t,
  title: n,
  width: N = 520,
  maskClosable: y = !0,
  footer: u,
  onClose: a,
  onOk: P,
  children: h,
  className: j,
  typeSpeed: A = 80,
  typewriter: B = !0,
  maskStyle: F
}) => {
  const [T, D] = $(0);
  c(() => {
    t && D((r) => r + 1);
  }, [t]);
  const f = k(null), b = k(null);
  c(() => {
    if (!t) return;
    b.current = document.activeElement ?? null;
    const r = window.setTimeout(() => {
      const i = f.current;
      if (!i) return;
      (E(i)[0] ?? i).focus();
    }, 0);
    return () => {
      window.clearTimeout(r), b.current?.focus?.();
    };
  }, [t]), c(() => {
    if (!t) return;
    const r = (i) => {
      if (i.key === "Escape") {
        a?.();
        return;
      }
      if (i.key !== "Tab") return;
      const l = f.current;
      if (!l) return;
      const d = E(l);
      if (d.length === 0) {
        i.preventDefault(), l.focus();
        return;
      }
      const g = d[0], L = d[d.length - 1], s = document.activeElement;
      i.shiftKey ? (s === g || !l.contains(s)) && (i.preventDefault(), L.focus()) : (s === L || !l.contains(s)) && (i.preventDefault(), g.focus());
    };
    return document.addEventListener("keydown", r), () => document.removeEventListener("keydown", r);
  }, [t, a]), c(() => (t && (document.body.style.overflow = "hidden"), () => {
    document.body.style.overflow = "";
  }), [t]);
  const I = w(() => {
    y && a?.();
  }, [y, a]), M = w((r) => {
    r.stopPropagation();
  }, []), p = `animal-modal-${O().replace(/:/g, "")}`, C = `${p}-title`, v = `${p}-body`;
  if (!t) return null;
  const R = /* @__PURE__ */ m(K, { children: [
    /* @__PURE__ */ e(x, { type: "primary", onClick: a, children: "取消" }),
    /* @__PURE__ */ e(x, { type: "primary", onClick: P, children: "确定" })
  ] }), S = /* @__PURE__ */ e(q, { children: /* @__PURE__ */ e("div", { className: o.mask, style: F, onClick: I, children: /* @__PURE__ */ m(
    "div",
    {
      ref: f,
      className: [o.modal, j].filter(Boolean).join(" "),
      style: { width: N },
      onClick: M,
      role: "dialog",
      "aria-modal": "true",
      "aria-labelledby": n ? C : void 0,
      "aria-describedby": v,
      tabIndex: -1,
      children: [
        /* @__PURE__ */ e(G, {}),
        /* @__PURE__ */ m("div", { className: o.modalClipped, children: [
          n && /* @__PURE__ */ e("div", { className: o.header, children: n && /* @__PURE__ */ e("div", { className: o.title, id: C, children: n }) }),
          /* @__PURE__ */ e("div", { className: o.body, id: v, children: B ? /* @__PURE__ */ e(_, { speed: A, trigger: T, children: h }) : h }),
          u !== null && /* @__PURE__ */ e("div", { className: o.footer, children: u === void 0 ? R : u })
        ] })
      ]
    }
  ) }) });
  return U(S, document.body);
};
H.displayName = "Modal";
export {
  H as Modal
};
