// animal-island-ui/dist/es/components/CodeBlock/CodeBlock.js
import { jsx as l } from "react/jsx-runtime";
const t = {
  comment: "#6b5e50",
  string: "#a8d4a0",
  keyword: "#d4a0e0",
  react: "#e06c75",
  component: "#80c0e0",
  func: "#61afef",
  prop: "#e8c87a",
  jsx: "#f0a870",
  operator: "#d4b896",
  number: "#a8d4a0",
  default: "#e8d5bc"
}, u = {
  padding: "20px 24px",
  background: "#2b2118",
  border: "1px solid #3d3028",
  borderRadius: 20,
  fontSize: 14,
  lineHeight: 1.7,
  fontFamily: "'SF Mono', 'Fira Code', 'Cascadia Code', Consolas, monospace",
  fontWeight: 600,
  color: "#e8d5bc",
  whiteSpace: "pre",
  overflow: "auto",
  tabSize: 4
}, f = (s) => {
  const a = [], e = (o, i) => {
    let c;
    const d = new RegExp(o.source, o.flags.includes("g") ? o.flags : o.flags + "g");
    for (; (c = d.exec(s)) !== null; )
      a.push({
        start: c.index,
        end: c.index + c[0].length,
        color: i
      });
  };
  e(/\/\*[\s\S]*?\*\//g, t.comment), e(/\/\/.*$/gm, t.comment), e(/`[^`]*`/g, t.string), e(/"[^"]*"/g, t.string), e(/'[^']*'/g, t.string), e(/<\/?[A-Z][\w.$]*/g, t.jsx), e(/<\/?[a-z][\w-]*/g, t.jsx), e(/\/?>/g, t.jsx), e(
    /\b(React|useState|useEffect|useCallback|useMemo|useRef|useContext|useReducer|useLayoutEffect|useImperativeHandle|useDebugValue|createContext|createElement|cloneElement|Fragment|Suspense|lazy|memo|forwardRef|useId|FC|ReactNode|ReactElement|CSSProperties)\b/g,
    t.react
  ), e(/\b(true|false)\b/g, t.keyword), e(/\b(null|undefined|void|NaN|Infinity)\b/gi, t.keyword), e(/\b\d+\.?\d*\b/g, t.number), e(
    /\b(import|from|as|export|default|const|let|var|function|return|if|else|for|while|switch|case|break|continue|try|catch|throw|finally|new|typeof|instanceof|async|await|type|interface)\b/gi,
    t.keyword
  ), e(/\b[A-Z][a-zA-Z0-9_$]*\b/g, t.component), e(/\b[a-z][a-zA-Z0-9_$]*\s*(?=\()/g, t.func), e(/\b[a-zA-Z_$][\w$]*\s*(?==)/g, t.prop), e(/>|===|!==|==|!=|<=|>=|&&|\|\||[+\-*/%=<>!&|^~?:]/g, t.operator), e(/[{}[\]();,]/g, t.operator), a.sort((o, i) => o.start - i.start);
  const r = [];
  let n = 0;
  for (const o of a)
    o.start < n || (o.start > n && r.push(
      /* @__PURE__ */ l("span", { style: { color: t.default }, children: s.slice(n, o.start) }, `t${n}`)
    ), r.push(
      /* @__PURE__ */ l("span", { style: { color: o.color }, children: s.slice(o.start, o.end) }, `s${o.start}`)
    ), n = o.end);
  return n < s.length && r.push(
    /* @__PURE__ */ l("span", { style: { color: t.default }, children: s.slice(n) }, `e${n}`)
  ), r;
}, p = ({ code: s, style: a, className: e }) => /* @__PURE__ */ l("pre", { style: { ...u, ...a }, className: e, children: f(s) });
export {
  p as CodeBlock
};
