// animal-island-ui/dist/es/components/Card/card.module.less.js
import "./card.module.css";
const a = "animal-card-DJ515", p = "animal-card-dashed-fqnm1", e = "animal-card-app-pink-3Eo8d", n = "animal-card-purple-Gq7sR", r = "animal-card-app-blue-GHH1a", t = "animal-card-app-yellow-DWB-L", l = "animal-card-app-orange-V5yD7", c = "animal-card-app-teal-R5kFG", d = "animal-card-app-green-xlNUs", i = "animal-card-app-red-DzE0l", m = "animal-card-lime-green-KiJbe", o = "animal-card-yellow-green-4c-lF", s = "animal-card-brown-hLCrx", w = "animal-card-warm-peach-pink-RlSVo", g = "animal-pattern-default-w-ord", u = "animal-pattern-app-pink--UNSj", A = "animal-pattern-purple-uTyc6", k = "animal-pattern-app-blue-KIXQM", h = "animal-pattern-app-yellow-K9MwO", b = "animal-pattern-app-orange-QdKbc", G = "animal-pattern-app-teal-bRRLd", P = "animal-pattern-app-green-FKDYa", y = "animal-pattern-app-red--c4q5", R = "animal-pattern-lime-green-z13TC", D = "animal-pattern-yellow-green-Cgkp8", f = "animal-pattern-brown-2fhlx", B = "animal-pattern-warm-peach-pink-EXPaC", K = {
  card: a,
  "card-dashed": "animal-card-dashed-fqnm1",
  cardDashed: p,
  "card-app-pink": "animal-card-app-pink-3Eo8d",
  cardAppPink: e,
  "card-purple": "animal-card-purple-Gq7sR",
  cardPurple: n,
  "card-app-blue": "animal-card-app-blue-GHH1a",
  cardAppBlue: r,
  "card-app-yellow": "animal-card-app-yellow-DWB-L",
  cardAppYellow: t,
  "card-app-orange": "animal-card-app-orange-V5yD7",
  cardAppOrange: l,
  "card-app-teal": "animal-card-app-teal-R5kFG",
  cardAppTeal: c,
  "card-app-green": "animal-card-app-green-xlNUs",
  cardAppGreen: d,
  "card-app-red": "animal-card-app-red-DzE0l",
  cardAppRed: i,
  "card-lime-green": "animal-card-lime-green-KiJbe",
  cardLimeGreen: m,
  "card-yellow-green": "animal-card-yellow-green-4c-lF",
  cardYellowGreen: o,
  "card-brown": "animal-card-brown-hLCrx",
  cardBrown: s,
  "card-warm-peach-pink": "animal-card-warm-peach-pink-RlSVo",
  cardWarmPeachPink: w,
  "pattern-default": "animal-pattern-default-w-ord",
  patternDefault: g,
  "pattern-app-pink": "animal-pattern-app-pink--UNSj",
  patternAppPink: u,
  "pattern-purple": "animal-pattern-purple-uTyc6",
  patternPurple: A,
  "pattern-app-blue": "animal-pattern-app-blue-KIXQM",
  patternAppBlue: k,
  "pattern-app-yellow": "animal-pattern-app-yellow-K9MwO",
  patternAppYellow: h,
  "pattern-app-orange": "animal-pattern-app-orange-QdKbc",
  patternAppOrange: b,
  "pattern-app-teal": "animal-pattern-app-teal-bRRLd",
  patternAppTeal: G,
  "pattern-app-green": "animal-pattern-app-green-FKDYa",
  patternAppGreen: P,
  "pattern-app-red": "animal-pattern-app-red--c4q5",
  patternAppRed: y,
  "pattern-lime-green": "animal-pattern-lime-green-z13TC",
  patternLimeGreen: R,
  "pattern-yellow-green": "animal-pattern-yellow-green-Cgkp8",
  patternYellowGreen: D,
  "pattern-brown": "animal-pattern-brown-2fhlx",
  patternBrown: f,
  "pattern-warm-peach-pink": "animal-pattern-warm-peach-pink-EXPaC",
  patternWarmPeachPink: B
};
export {
  a as card,
  r as cardAppBlue,
  d as cardAppGreen,
  l as cardAppOrange,
  e as cardAppPink,
  i as cardAppRed,
  c as cardAppTeal,
  t as cardAppYellow,
  s as cardBrown,
  p as cardDashed,
  m as cardLimeGreen,
  n as cardPurple,
  w as cardWarmPeachPink,
  o as cardYellowGreen,
  K as default,
  k as patternAppBlue,
  P as patternAppGreen,
  b as patternAppOrange,
  u as patternAppPink,
  y as patternAppRed,
  G as patternAppTeal,
  h as patternAppYellow,
  f as patternBrown,
  g as patternDefault,
  R as patternLimeGreen,
  A as patternPurple,
  B as patternWarmPeachPink,
  D as patternYellowGreen
};
