// animal-island-ui/dist/es/index.js
/* empty css                  */
import { Button as t } from "./components/Button/Button.js";
import { Card as p } from "./components/Card/Card.js";
import { Checkbox as f } from "./components/Checkbox/Checkbox.js";
import { CodeBlock as l } from "./components/CodeBlock/CodeBlock.js";
import { Collapse as T } from "./components/Collapse/Collapse.js";
import { Cursor as C } from "./components/Cursor/Cursor.js";
import { Divider as n } from "./components/Divider/Divider.js";
import { Footer as s } from "./components/Footer/Footer.js";
import { Form as F } from "./components/Form/Form.js";
import { FormItem as h } from "./components/Form/FormItem.js";
import { ICON_LIST as k, Icon as w } from "./components/Icon/Icon.js";
import { Input as L } from "./components/Input/Input.js";
import { Loading as v } from "./components/Loading/Loading.js";
import { Modal as D } from "./components/Modal/Modal.js";
import { Phone as N } from "./components/Phone/Phone.js";
import { Radio as P } from "./components/Radio/Radio.js";
import { Select as W } from "./components/Select/Select.js";
import { Switch as j } from "./components/Switch/Switch.js";
import { Table as z } from "./components/Table/Table.js";
import { Tabs as E } from "./components/Tabs/Tabs.js";
import { Time as H } from "./components/Time/Time.js";
import { Title as K } from "./components/Title/Title.js";
import { Tooltip as U } from "./components/Tooltip/Tooltip.js";
import { Typewriter as X } from "./components/Typewriter/Typewriter.js";
import { Wallet as Z } from "./components/Wallet/Wallet.js";
import { useForm as oo } from "./components/Form/useForm.js";
export {
  t as Button,
  p as Card,
  f as Checkbox,
  l as CodeBlock,
  T as Collapse,
  C as Cursor,
  n as Divider,
  s as Footer,
  F as Form,
  h as FormItem,
  k as ICON_LIST,
  w as Icon,
  L as Input,
  v as Loading,
  D as Modal,
  N as Phone,
  P as Radio,
  W as Select,
  j as Switch,
  z as Table,
  E as Tabs,
  H as Time,
  K as Title,
  U as Tooltip,
  X as Typewriter,
  Z as Wallet,
  oo as useForm
};
