// animal-island-ui/dist/es/components/Divider/divider.module.less.js
import "./divider.module.css";
const e = "animal-divider-ZBhpE", l = "animal-line-teal-Z80hb", a = "animal-line-white-VDY1E", i = "animal-line-yellow-ifvJz", d = "animal-wave-yellow-AO-io", n = "animal-dashed-brown--I1Lx", h = "animal-dashed-teal-K-4mV", o = "animal-dashed-white-iMYTR", s = "animal-dashed-yellow-Vu6Tq", w = {
  divider: e,
  "line-teal": "animal-line-teal-Z80hb",
  lineTeal: l,
  "line-white": "animal-line-white-VDY1E",
  lineWhite: a,
  "line-yellow": "animal-line-yellow-ifvJz",
  lineYellow: i,
  "wave-yellow": "animal-wave-yellow-AO-io",
  waveYellow: d,
  "dashed-brown": "animal-dashed-brown--I1Lx",
  dashedBrown: n,
  "dashed-teal": "animal-dashed-teal-K-4mV",
  dashedTeal: h,
  "dashed-white": "animal-dashed-white-iMYTR",
  dashedWhite: o,
  "dashed-yellow": "animal-dashed-yellow-Vu6Tq",
  dashedYellow: s
};
export {
  n as dashedBrown,
  h as dashedTeal,
  o as dashedWhite,
  s as dashedYellow,
  w as default,
  e as divider,
  l as lineTeal,
  a as lineWhite,
  i as lineYellow,
  d as waveYellow
};
